### DISCUSSION TITLE
Should i enable kernel-mode driver?

### DISCUSSION THREAD

**Discussion by TeusDev** [winsiderss/systeminformer#1375] [Questions / Answers] [answered]
What is the benefit by switch on kernel-mode driver ? i think that the button miss somekind of a explanation bubble or something.

**Comment by Aholicknight** [comment]
FYI if you update to the recent Windows 11 update this driver signature (for some reason) is now invalid
![image](https://user-images.githubusercontent.com/7843719/191827156-d991cc47-0f48-488d-a3bf-29ea809b1833.png)

**Comment by God-damnit-all** [comment]
@dmex This was never answered, I would also like to know what benefits the kernel-mode driver provides, I can't find any documentation on it.

**Comment by jxy-s** [comment, marked as answer]
It's touched on here: https://github.com/winsiderss/systeminformer/blob/master/KSystemInformer/README.md

Mostly the driver enables deeper access into information that is not normally exposed by the kernel. In other situations it enables you to manipulate processes that you would otherwise be unable to. And finally it exposes APIs that provide faster queries for information that would otherwise involve many enumerations or duplication of effort.

Processes: https://github.com/winsiderss/systeminformer/blob/master/kphlib/include/kphapi.h#L100-L116
Threads: https://github.com/winsiderss/systeminformer/blob/master/kphlib/include/kphapi.h#L118-L133
Objects: https://github.com/winsiderss/systeminformer/blob/master/kphlib/include/kphapi.h#L118-L133
Drivers: https://github.com/winsiderss/systeminformer/blob/master/kphlib/include/kphapi.h#L274-L281
ALPC Ports: https://github.com/winsiderss/systeminformer/blob/master/kphlib/include/kphapi.h#L310-L315
System: https://github.com/winsiderss/systeminformer/blob/master/kphlib/include/kphapi.h#L364-L368

The next driver release will:
- Enable SI to associate WSLv1 pico processes: https://github.com/winsiderss/systeminformer/blob/kph-staging/kphlib/include/kphapi.h#L87
- Expose and display thread I/O counters: https://github.com/winsiderss/systeminformer/blob/kph-staging/kphlib/include/kphapi.h#L105
- Extend visibility of file/image mappings: https://github.com/winsiderss/systeminformer/blob/kph-staging/kphlib/include/kphapi.h#L355-L358

This is a complete list of changes coming to the driver next: https://github.com/winsiderss/systeminformer/compare/master...kph-staging

Personally, the piece I find most useful in the current features of the driver is the extended information on ALPC ports, best viewed in the handles window:
![image](https://user-images.githubusercontent.com/11687482/235033935-f6883bf7-f3e8-4067-aa92-f078b6d36606.png)

Without the driver active you don't get the information showing details about the client/connection/server.

**Reply by God-damnit-all** [reply]
I see! Unfortunately, after I enabled it, System Informer started lagging so bad that it took about 10 minutes to disable it again just due to how long it took for the GUI to respond.
